mora/validator.py

```diff
@@ -6,14 +6,18 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 import datetime
+import typing
+
 from . import exceptions
 from . import lora
 from . import util
+
 from .service import common
 from .service import keys
 
 
-def _is_date_range_valid(parent: str, startdate: datetime.datetime,
+def _is_date_range_valid(parent: typing.Union[dict, str],
```

Comment:
Update docstring to reflect what `parent` is now.
And maybe we should reduce this to only being `dict`, seeing as that's the only thing we use it for now.

Reply:
Well, we pass it on to `get_effects()` which takes either, so the type annotation should be correct — and we might want to reuse ID support later.

I've fixed the documentation.

Reply:
That's true. I wanted to remove the use of the scope parameter as we don't really need to fetch the object, but I realized that's where `get_effects` lives.